scio-core/src/main/scala/com/spotify/scio/schemas/Schema.scala

```diff
@@ -139,7 +139,7 @@ object LogicalType {
     }
   }
 
-  def unapply[T](logicalType: LogicalType[T]): Option[BSchema.FieldType] =
+  def unapply[T](logicalType: LogicalType[T]): Some[BSchema.FieldType] =
```

Comment:
if it returns an `Option`, the compiler cannot make sure of pattern match completeness and will issue warnings that some case may be unhanded. Using `Some` it will be considered the same as `case _: LogicalType[T]` as it guarantees any logical type will get his field extracted

Reply:
Ah, I was under the incorrect impression that `unapply` methods had a more stable contract (i.e. required `Option`) 👍